### PR TITLE
[chore] refactor block modification handler to separate class

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Mv to separate class so that we can inject and reuse more easily.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/routes/block_modification_checker.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+"""
+Block modification checker
+"""
+from typing import Iterator
+
+from contextlib import contextmanager
+
+
+class BlockModificationChecker:
```
Comment (gee-senbong) [inline featurebyte/routes/block_modification_checker.py:9]: `Checker` sounds like it only perform checking related tasks. But this class actually control block modification behaviour (either enable or disable the check).

Comment (jevonyeoh) [reply featurebyte/routes/block_modification_checker.py:9]: hm happy to rename, maybe to `BlockModificationHandler`? any suggestions? 🤔

Comment (gee-senbong) [reply featurebyte/routes/block_modification_checker.py:9]: Handler seems better to me 👍

**Review by gee-senbong — APPROVED** [review]
LGTM

**Pull request merged**